pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -779,7 +779,9 @@ private void handleOwnEvent(String serviceUnit, ServiceUnitStateData data) {
             lastOwnEventHandledAt = System.currentTimeMillis();
             stateChangeListeners.notify(serviceUnit, data, null);
             log(null, serviceUnit, data, null);
-        } else if ((data.force() || isTransferCommand(data)) && isTargetBroker(data.sourceBroker())) {
+        } else if (isTargetBroker(data.sourceBroker())
+                && (data.force() // orphan cleanup
+                || (isTransferCommand(data) && pulsar.getConfig().isLoadBalancerMultiPhaseBundleUnload()))) { // txfer
             stateChangeListeners.notifyOnCompletion(
                             closeServiceUnit(serviceUnit, true), serviceUnit, data)
```

Comment:
I am sorry. I realized We still need to log the transfer case even when this flag is disabled.

I think we need to revert this if-logic, but instead pass the Completed future when this flag is disabled inside the block.